libbeat/monitoring/adapter/filters.go

```diff
@@ -90,18 +91,43 @@ func WhitelistIf(pred func(string) bool) MetricFilter {
 	return ApplyIf(pred, Accept)
 }
 
-// Whitelist sets a list of metric names to be accepted.
-func Whitelist(names ...string) MetricFilter {
-	return WhitelistIf(func(name string) bool {
+// NameIn checks a metrics name matching any of the set names
+func NameIn(names ...string) func(string) bool {
+	return func(name string) bool {
 		for _, n := range names {
 			if name == n {
 				return true
 			}
 		}
 		return false
+	}
+}
+
+// Whitelist sets a list of metric names to be accepted.
+func Whitelist(names ...string) MetricFilter {
+	return WhitelistIf(NameIn(names...))
+}
+
+// Mode sets the monitoring mode (Basic, Enhanced, Full) for
```

Comment:
We had a similar concept in the beginning of metricbeat on which metrics should be sent to elasticsearch for a metricset. We dropped the idea rather soon because the disussion what should be in basic, enhanced and full got too complex as the answer was different for every single person. I would suggest we follow here more the approach of going with whitelist / backlist filtering instead.